source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -0,0 +1,325 @@
+        {
+            vec4 outColor = inPixel;
+
+            // Add a Matrix processing
+
+            outColor = mat4(1.7049986809609248, -0.13025866763271018, -0.024007514741958086, 0., -0.62174765641381313,
+            1.1408020008115505, -0.12898266957287285, 0., -0.083251260888346112, -0.010544731974266384, 1.1529906045288774, 0., 0., 0., 0., 1.)
+            * outColor;
+
+            return outColor;
+        }
+        */
+
+        // TODO : Create an implementation based on source code
+        // Note: A separate utility could take OCIO shader code and generate a set of MTLX implementations.
+        //       This could be useful to update the DefaultColorSystems implementations to maintain 
+        //       consistency with OCIO.
+        ImplementationPtr impl = _document->addImplementation(transformFunctionName);
+        if (impl)
+        {
+            // Q: How to parse inputs ?
+            impl->addInput("inPixel", transform.type->getName());
+            impl->addOutput(outputName, transform.type->getName());
+            impl->setAttribute("sourcecode", fullFunction);
+            impl->setAttribute("function", functionName);
+            impl->setAttribute("target", _ocioInfo->target);
+        }
+        return impl;
+    }
+    catch (const OCIO::Exception& e)
+    {
+        throw ExceptionShaderGenError("Code generation failed for transform: '" + 
+            transform.sourceSpace + "' to '" + transform.targetSpace + "'. Type: '" + transform.type->getName()  + "' : " + e.what());
+    }
+
+    return nullptr;
+}
+
+//
+// OCIO source code node method
+//
+
+ShaderNodeImplPtr OCIOSourceCodeNode::create()
+{
+    return std::make_shared<OCIOSourceCodeNode>();
+}
+
+void OCIOSourceCodeNode::emitFunctionDefinition(const ShaderNode&, GenContext& context, ShaderStage& stage) const
+{
+    // Write source code as is
+    BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
+    if (!_functionSource.empty())
+    {
+        const ShaderGenerator& shadergen = context.getShaderGenerator();
+        shadergen.emitBlock(_functionSource, context, stage);
+        shadergen.emitLineBreak(stage);
+    }
+    END_SHADER_STAGE(stage, Stage::PIXEL)
+}
+
+void OCIOSourceCodeNode::emitFunctionCall(const ShaderNode& node, GenContext& context, ShaderStage& stage) const
+{
+    // TODO: Put in correct code here. Just a copy of HwSourceCode
+    BEGIN_SHADER_STAGE(stage, Stage::PIXEL)
+    if (_inlined)
+    {
+        SourceCodeNode::emitFunctionCall(node, context, stage);
+    }
+    else
+    {
+        const ShaderGenerator& shadergen = context.getShaderGenerator();
+
+        // Declare the output variable
+        shadergen.emitLineBegin(stage);
+        shadergen.emitOutput(node.getOutput(0), true, false, context, stage);
+        shadergen.emitString(" = ", stage);
+
+        // Emit function name. TODO add vec3/vec4 conversion as needed
+        shadergen.emitString(_functionName + "(", stage);
```

Comment:
need to wrap this into a vec3->vec4(ocio_function)->vec3 call if this is being called to return a vec3 value.